.github/workflows/stats-comment.yml

```diff
@@ -14,32 +14,115 @@ jobs:
     if: ${{ github.event.pull_request && !github.event.pull_request.head.repo.fork }}
 
     steps:
-      - name: Checkout code
+      - name: Cache stats
+        uses: actions/cache@v4.2.3
+        id: stats-cache
+        with:
+          # Restore build cache (unless commit SHA changes)
+          key: build-stats-${{ github.sha }}
```

Comment:
We might want to include the `base_ref` in the hash key as well, as we'll want to re-compute the stats if the target branch changes.